float32 framerate
---
bool ret
